sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -186,11 +197,17 @@ private boolean shouldApplyScopeData(
 
     final List<SentryEnvelopeItem> envelopeItems = new ArrayList<>();
 
+    Boolean isFatal = false;
+
     if (event != null) {
       final SentryEnvelopeItem eventItem =
           SentryEnvelopeItem.fromEvent(options.getSerializer(), event);
       envelopeItems.add(eventItem);
       sentryId = event.getEventId();
+      if (event instanceof SentryEvent) {
```

Comment:
The idea here was to attach the current counts to a fatal event as that's what is recorded on a crash. This way if the event still makes it to the store we can restore the counts and send them later.

Reply:
I'm not entire sure about this, let's talk about this on standup.